lib-python/littleboxes/solver/cluedb_solver.py

```diff
@@ -1,15 +1,225 @@
 from littleboxes.solver.solver import Solver
+from littleboxes.xword import InvalidCrosswordException
+
+from copy import deepcopy
+import itertools
+import time
 
 class ClueDBSolver(Solver):
     """Solver that looks for clue matches in the provided database."""
+
     def __init__(self, db, distance_cutoff=0):
         """Args:
             db (ClueDB): The database of clues to search for answers.
-            distance_cutoff (int): Clues will only be considered a match if they
-                have Levenstein distance <= cutoff. Default is 0 meaning exact match.
+            distance_cutoff (int): Clues will only be considered a match if 
+                they have Levenstein distance <= cutoff. Default is 0 meaning
+                exact match.
         """
         self._db = db
         self._distance_cutoff = distance_cutoff
 
     def solve(self, xword):
-        pass
+        '''Set-based search for partial solutions to a crossword
+
+        This implementation is highly speed-efficiently, and very memory
+        efficient (although there could be improvements to the latter by not
+        loading certain things into memory at once)
+
+        Precomputes possible answers to every clue using query_answers and
+        determines what conflicts exist between across and down answers first
+
+        Arguments:
+            xword - a Crossword to be solved
+
+        Returns:
+            sorted list of tuples (Crossword, prob) of partially solved 
+            Crosswords in order from most- to least-solved with prob=1.0 for
+            each Crossword for now
+        '''
+
+        possible_answers = self.query_answers(xword)
+
+        start_conflicts = time.time()
+        conflict_map = self.build_conflict_map(xword, possible_answers)
+
+        print 'Conflicts map build in {0} seconds'.format(
```

Comment:
Yeah!  ClueDBSolver doesn't have a logger to call it's own, should I just use the root logger with the default formatting?  i.e. logging.info('stuff')